### PR TITLE
Update RAGAgentLlama and ReActLlama

### PR DESCRIPTION
**Pull request by minmin-intel**
## Description

1. Use ChatOpenAI to interface with TGI-gaudi as ChatHuggingFace has a bug
2. Update tools for unit test as duckduckgo throws strange rate limit error

## Issues
 `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds new functionality)

## Dependencies

No newly introduced 3rd party dependency.

## Tests

unit test

### CODE REVIEW

File: comps/agent/langchain/src/utils.py [hunk 1]
```diff
@@ -99,6 +99,19 @@ def setup_llm(args):
     return model
 
 
+def setup_chat_model(args):
```
Comment (xuechendi) [inline comps/agent/langchain/src/utils.py:102]: Looks this function is same as setup_vllm_client, can we merge as one?

Comment (minmin-intel) [reply comps/agent/langchain/src/utils.py]: yes. I updated the code.

File: comps/agent/langchain/src/strategy/ragagent/planner.py [hunk 2]
```diff
@@ -270,16 +262,19 @@ class QueryWriterLlama:
...
+        self.tools = tools
+        self.args = args
+
         output_parser = QueryWriterLlamaOutputParser()
         prompt = PromptTemplate(
             template=QueryWriterLlamaPrompt,
             input_variables=["question", "history", "feedback"],
         )
-        llm = ChatHuggingFace(llm=llm_endpoint, model_id=model_id)
+        llm = setup_chat_model(args)
```
[the remark this answers is not in the document]
Comment (minmin-intel) [reply comps/agent/langchain/src/strategy/ragagent/planner.py:186]: Yes. I updated the code.

**Review by xuechendi — APPROVED** [review]
LGTM

**Review by lkk12014402 — APPROVED** [review]
LGTM

**Comment by minmin-intel** [comment]
Tested with OpenAI GPT-4o-mini for react_langgraph and ragagent. Screenshots below.
image

**Pull request merged**